models/default_local_provider.go

```diff
@@ -507,12 +508,12 @@ func (l *DefaultLocalProvider) RemotePatternFile(req *http.Request, resourceURL,
 	// Check if hostname is github
 	if parsedURL.Host == "github.com" {
 		parsedPath := strings.Split(parsedURL.Path, "/")
-		if parsedPath[3] == "tree" {
-			parsedPath = append(parsedPath[0:3], parsedPath[4:]...)
-		}
 		if len(parsedPath) < 3 {
 			return nil, fmt.Errorf("malformed URL: url should be of type github.com/<owner>/<repo>/[branch]")
 		}
+		if len(parsedPath) >= 4 && parsedPath[3] == "tree" {
```

Comment:
Do we have a common function for https://github.com/<org>/<repo>/releases/latest?